frontend/src/components/pages/CampOverview/RemoveCamperModal/SelectModal.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+
+import {
+  Box,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  Button,
+  Text,
+  Checkbox,
+  Stack,
+} from "@chakra-ui/react";
+
+import { Camper } from "../../../../types/CamperTypes";
+
+type SelectModalProps = {
+  camper: Camper;
+  campersToBeDeleted: Set<Camper>;
+  setCampersToBeDeleted: React.Dispatch<React.SetStateAction<Set<Camper>>>;
+  retrievedCampers: Camper[];
+  deselectAndClose: () => void;
+  setStatusAsConfirm: () => void;
+};
+
+const SelectModal = ({
+  camper,
+  campersToBeDeleted,
+  setCampersToBeDeleted,
+  retrievedCampers,
+  deselectAndClose,
+  setStatusAsConfirm,
+}: SelectModalProps): JSX.Element => {
+  const addCamperToBeDeleted = (newCamper: Camper) => {
+    setCampersToBeDeleted(new Set(campersToBeDeleted.add(newCamper)));
+  };
+
+  const removeCamperToBeDeleted = (newCamper: Camper) => {
+    setCampersToBeDeleted(
+      new Set(
+        (Array.from(campersToBeDeleted) as Camper[]).filter(
+          (currentCamper: Camper) => currentCamper.id !== newCamper.id,
+        ),
+      ),
+    );
+  };
```

Comment:
Would this work? It might not work because of objects and all. Also nit, maybe let's rename it from `newCamper`?

```suggestion
const removeCamperToBeDeleted = (newCamper: Camper) => {
    const newSet = new Set(campersToBeDeleted);
    newSet.remove(newCamper);
    setCampersToBeDeleted(newSet);
  };
```

Reply:
Yeah lol that's a lot cleaner, it does still work since we're still changing the entire object in the state.